### ISSUE TITLE
[BUG] TypeError while using "pyarmor gen key -e 3" command

### ISSUE THREAD

**Issue by MatTZer28** [dashingsoft/pyarmor#1057]
I was following section 1.3.5 of the document, I attempted to generate an outer runtime key after the successful completion of "pyarmor gen --outer foo.py". I then typed the command "pyarmor gen key -e 3", but it didn't work and showed a TypeError.

Executed command:
```
pyarmor gen key -e 3
```
Output:
```
INFO     Python 3.11.2
INFO     Pyarmor 8.0.25 (trial), 000000, non-profits
INFO     start to generate outer runtime key "None"
CRITICAL unknown error, please check pyarmor.error.log
ERROR    join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```
pyarmor.error.log:
```
18252 MainProcess 2023-03-31 15:44:24,723
Traceback (most recent call last):
  File "C:\Users\qazws\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyarmor\cli\__main__.py", line 601, in main
    main_entry(sys.argv[1:])
  File "C:\Users\qazws\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyarmor\cli\__main__.py", line 589, in main_entry
    args.func(ctx, args)
  File "C:\Users\qazws\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyarmor\cli\__main__.py", line 148, in cmd_gen
    _cmd_gen_key(builder, options)
  File "C:\Users\qazws\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyarmor\cli\__main__.py", line 47, in _cmd_gen_key
    target = os.path.join(output, keyname)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen ntpath>", line 147, in join
  File "<frozen genericpath>", line 152, in _check_arg_types
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'NoneType'
```

File structure:
```
pyarmor-demo/
└── foo.py
└── dist/
```

**Comment by jondy** [comment]
Thanks, it's bug.

I have pushed fixed code
be00a5c11b29e1aba55bb1b73c77df504cdd2026

**Comment by jondy** [comment]
Fixed in 8.0.26
